packages/frontend-2/lib/viewer/composables/serialization.ts

```diff
@@ -50,7 +50,7 @@ export function useStateSerialization() {
   ): SerializedViewerState => {
     const { concreteResourceIdString } = options || {}
 
-    const camControls = state.viewer.instance.getExtension(CameraController).controls
+    const camControls = state.viewer.instance.getExtension(CameraController)!.controls
```

Comment:
@AlexandruPopovici 

Every time you use `!` or `xx as SomeType` you're essentially overriding TypeScript's typing system and potentially introducing a location in code where an error can occur (just like they've been happening so far) due to the object not existing or being the wrong type.

You should use these as sparingly as possible, and instead change your code to explicitly check for the correct type through type guards or for existence of properties.

I'd even go as far as to mark `!` usages as an error through ESLint (we have that on the server side)